File: Programmers/py/lev2/귤고르기/Solution.py
from collections import Counter
def solution(k, tangerine):
    cnt = 0 
    
    dic = dict()
    for i in tangerine:
        dic[i] = dic.get(i,0)+1
    lst = sorted(dic.values() , reverse =True)
    # lst = sorted(Counter(tangerine).values(),reverse=True)
    
    i = 0 
    while(k > 0):
        k -= lst[i]
        cnt += 1
        i+=1
    
    return cnt

File: Programmers/java/lev2/귤고르기/Solution.java
import java.util.Arrays;
class Solution{
    public int solution(int k, int[] tangerine) {
        int cnt = 0;
        Arrays.sort(tangerine);
        int[] lst = new int[ tangerine[tangerine.length-1] ];

        for (int i : tangerine)
            lst[i-1] ++;
        
        Arrays.sort(lst);
        
        for (int i = lst.length-1 ; i>=0; i--){
            k -= lst[i];
            cnt ++;
            
            if( k <= 0)
                break;
        }
        
        return cnt;
    }

    public static void main(){
        Solution sol = new Solution();
        int k = 6;
        int [] tangerine = new int[]{1, 3, 2, 5, 4, 5, 2, 3};
        sol.solution(k, tangerine);
    }
}